tests/testthat/test-rstudio.R

```diff
@@ -24,27 +24,30 @@ test_that("use_rstudio() omits package-related config for a project", {
   expect_true(is.na(match("BuildType: Package", out)))
 })
 
-test_that("a non-RStudio project is not recognized", {
-  create_local_package(rstudio = FALSE)
-  expect_false(is_rstudio_project())
-  expect_identical(rproj_path(), NA_character_)
-})
-
 test_that("an RStudio project is recognized", {
   create_local_package(rstudio = TRUE)
   expect_true(is_rstudio_project())
   expect_match(rproj_path(), "\\.Rproj$")
 })
 
-test_that("we error for multiple Rproj files", {
-  create_local_package(rstudio = TRUE)
-  file_copy(
-    rproj_path(),
-    proj_path("copy.Rproj")
-  )
-  expect_usethis_error(rproj_path(), "Multiple .Rproj files found", fixed = TRUE)
+test_that("we error if there isn't exactly one Rproj files", {
+  dir <- withr::local_tempdir()
+  path <- dir_create(path(dir, "test"))
+
+  expect_snapshot(rproj_path(path), error = TRUE)
+
+  file_touch(path(path, "a.Rproj"))
+  file_touch(path(path, "b.Rproj"))
+  expect_snapshot(rproj_path(path), error = TRUE)
 })
 
+test_that("a non-RStudio project is not recognized", {
+  create_local_package(rstudio = FALSE)
+  expect_false(is_rstudio_project())
+  expect_error(rproj_path(), NA_character_)
```

Comment:
This doesn't seem right. We *should* get an error, yeah? I actually don't understand why tests and CI are passing.

Reply:
If I run the whole test, it passes. But not if I walk through it line by line.

```
> devtools::load_all(".")
ℹ Loading usethis
> 
> test_that("a non-RStudio project is not recognized", {
+   create_local_package(rstudio = FALSE)
+   expect_false(is_rstudio_project())
+   expect_error(rproj_path(), NA_character_)
+ })
v Setting active project to '/private/tmp/Rtmpo0bGHz/testpkgee6f50e07b7b'
v Restoring original working directory: '/Users/jenny/rrr/usethis/'
v Setting active project to '<no active project>'
v Deleting temporary project: '/tmp/Rtmpo0bGHz/testpkgee6f50e07b7b/'
Test passed 😸
>   create_local_package(rstudio = FALSE)
Setting global deferred event(s).
i These will be run:
  * Automatically, when the R session ends.
  * On demand, if you call `withr::deferred_run()`.
i Use `withr::deferred_clear()` to clear them without executing.
✔ Setting active project to '/private/tmp/Rtmpo0bGHz/testpkgee6f348ba07'
>   expect_false(is_rstudio_project())
>   expect_error(rproj_path(), NA_character_)
Error in if (.p(.cnd, ...)) { : missing value where TRUE/FALSE needed
```

Reply:
Yeah, something does not make sense here. If I change this to `expect_snapshot()`, I definitely get an error.

```
test_that("a non-RStudio project is not recognized", {
  create_local_package(rstudio = FALSE)
  expect_false(is_rstudio_project())
  expect_snapshot(rproj_path())
  #expect_error(rproj_path(), NA_character_)
})
```

which gives

```
> devtools:::test_active_file()
[ FAIL 1 | WARN 0 | SKIP 0 | PASS 19 ]

── Error (test-rstudio.R:47): a non-RStudio project is not recognized ──────────
`rproj_path()` threw an unexpected error.
Message: "testpkgee6f39d74a7" is not an RStudio Project.
Class:   rlang_error/error/condition
Backtrace:

[ FAIL 1 | WARN 0 | SKIP 0 | PASS 19 ]
```